Day12/src/controllers/studentController.ts

```diff
@@ -0,0 +1,388 @@
+const uuid = require('uniqid');
+const services = require('../services/studentServices');
+
+exports.addStudent = async (ctx: any) => {
+
+
+    let object = ctx.request.body;
+    if (object.name == null || object.gender == null || object.phone == null || object.classId == null) {
+        ctx.response.status = 400;
+        ctx.response.type = 'application/json';
+        ctx.body = {
+            "msg": "Data missing"
+        };
+        return;
+    }
+
+    try {
+
+        let id = uuid('S')
+        await services.addStudent(id, object.name, object.gender, object.phone, object.classId);
+        ctx.response.status = 200;
+        ctx.response.type = 'application/json';
+        ctx.body = {
+            "msg": "Student added"
+        };
+
+
+    }
+    catch (e) {
+        console.log(e.stack);
+        ctx.response.status = 400;
+        ctx.response.type = 'application/json';
+        ctx.body = {
+            "msg": "something wrong happens"
+        };
+
+    }
+
+
+}
+exports.addTeacher = async (ctx: any) => {
+
+
+    let object = ctx.request.body;
+    if (object.name == null || object.gender == null || object.phone == null || object.subjectId == null) {
+        ctx.response.status = 400;
+        ctx.response.type = 'application/json';
+        ctx.body = {
+            "msg": "Data missing"
+        };
+        return;
+    }
+
+    try {
+
+        let id = uuid('T')
+        await services.addTeacher(id, object.name, object.gender, object.phone, object.subjectId);
+        ctx.response.status = 200;
+        ctx.response.type = 'application/json';
+        ctx.body = {
+            "msg": "Teacher added"
+        };
+
+
+    }
+    catch (e) {
+        console.log(e.stack);
+        ctx.response.status = 400;
+        ctx.response.type = 'application/json';
+        ctx.body = {
+            "msg": "something wrong happens"
+        };
+
+    }
+
+
+}
+exports.addSubject = async (ctx: any) => {
```

Comment:
separate controller for this. Lets create controller based resource they are handing

Reply:
can you do the same for others also